library/core/src/lib.rs

```diff
@@ -289,6 +289,15 @@ pub mod assert_matches {
     pub use crate::macros::{assert_matches, debug_assert_matches};
 }
 
+#[cfg(not(bootstrap))]
+#[unstable(feature = "autodiff", issue = "124509")]
+mod autodiff {
+    pub use crate::macros::builtin::autodiff;
+}
+#[cfg(not(bootstrap))]
+#[unstable(feature = "autodiff", issue = "124509")]
+pub use autodiff::autodiff;
```

Comment:
Discussion: the libs changes I'm not too sure about, is this needed for `#[autodiff]` for name resolution purposes? But this is not provided through the prelude right?